### PR TITLE
Enable checking headers with Clang Tidy.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/134969.

Turns out this was literally 1 line. Tested locally:

```shell
dart tools/clang_tidy/bin/main.dart \
  --target-variant host_debug_unopt_arm64 \
  --lint-all \
  --checks="-*,llvm-header-guard"
```

<details>

<summary>Local Test Results</summary>

Here is a tiny snippet showing this is indeed doing something:

```txt
[0:02] Jobs:   0% done,   0/1033 completed,  7 in progress, 1019 pending,   7 failed.    
❌ Failures for clang-tidy on /Users/matanl/Developer/engine/src/flutter/display_list/dl_blend_mode.cc:
../../flutter/display_list/dl_blend_mode.h:5:9: error: header guard does not follow preferred style [llvm-header-guard,-warnings-as-errors]
    5 | #ifndef FLUTTER_DISPLAY_LIST_DL_BLEND_MODE_H_
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         USERS_MATANL_DEVELOPER_ENGINE_SRC_OUT_HOST_DEBUG_UNOPT_ARM64_______FLUTTER_DISPLAY_LIST_DL_BLEND_MODE_H
    6 | #define FLUTTER_DISPLAY_LIST_DL_BLEND_MODE_H_
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         USERS_MATANL_DEVELOPER_ENGINE_SRC_OUT_HOST_DEBUG_UNOPT_ARM64_______FLUTTER_DISPLAY_LIST_DL_BLEND_MODE_H
^C
```

</details>

This will likely show failures on CI, so I'm following [our published directions](https://github.com/flutter/flutter/wiki/Engine-Clang-Tidy-Linter#clang-tidy-fix-on-ci) to get CI to tell me what needs to get fixed, and won't submit until there are no warnings at HEAD. As such, not seeking LGTM yet (draft).

### CODE REVIEW

**Comment by matanlurey** [comment]
I need to revert the CI script but I wanted to show the CI logs as-is as part of the review!

File: .clang-tidy [hunk 1]
```diff
@@ -39,3 +39,11 @@ CheckOptions:
     value: "lower_case"
   - key: readability-identifier-naming.PrivateMemberSuffix
     value: "_"
+
+# Lint headers within paths contain "/flutter/" but not:
```
Comment (zanderso) [inline .clang-tidy:43]:
```suggestion
# Lint headers within paths that contain "/flutter/" but not:
```

**Comment by zanderso** [comment]
Sorry, I'm a little confused about the state we're in. In the CI logs, it doesn't look like there are any lints reported in header files. Does that mean that they're all fixed already?

**Comment by matanlurey** [comment]
> Sorry, I'm a little confused about the state we're in. In the CI logs, it doesn't look like there are any lints reported in header files. Does that mean that they're all fixed already?

Yup. It took about 20 PRs but we are (currently) green as far as we can tell.

**Comment by zanderso** [comment]
> Yup. It took about 20 PRs but we are (currently) green as far as we can tell.

Whoa. That's awesome!

**Review by zanderso — APPROVED** [review]
LGTM w/ noted cleanup.

**Pull request merged**